composeApp/src/commonMain/kotlin/com/majotyler/hiittimer/presentation/common/navigation/Router.kt

```diff
@@ -0,0 +1,9 @@
+package com.majotyler.hiittimer.presentation.common.navigation
+
+/**
+ * A [Router] abstracts requests to route to a given [Destination]. It is expected that a child
+ * class of [Destination] is used instead of directly using [Destination].
+ */
+fun interface Router<Destination>  {
```

Comment:
A `fun interface` is a type of interface called a "functional interface". It's not that important, but you use it when you have an interface that only has a single function: like how this only has `routeTo`. It would let you do this `Router { }` instead of `object : Router { }`